src/Common/tests/System/Net/RemoteServerQuery.cs

```diff
@@ -0,0 +1,79 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.Net.Test.Common
+{
+    internal static class RemoteServerQuery
+    {
+        internal static async Task Run(Func<Task> testCode, Type remoteServerFailExceptionType, string servername, bool mustFail = true)
```

Comment:
Why `mustFail` at all?

Reply:
There can be tests which we may want to pass

Reply:
I am not convinced we will ever want that. At minimum we should delay introducing something like this until we really need it.